python/pyspark/sql/functions.py

```diff
@@ -2131,6 +2131,32 @@ def weekofyear(col: "ColumnOrName") -> Column:
     return Column(sc._jvm.functions.weekofyear(_to_java_column(col)))
 
 
+def make_date(year: "ColumnOrName", month: "ColumnOrName", day: "ColumnOrName") -> Column:
+    """
+    Returns a column with a date built from the year, month and day columns.
+
+    .. versionadded:: 3.3.0
+
+    Parameters
+    ----------
+    year : :class:`ColumnOrName` with the year to build the date
```

Comment:
Type hints should be `ColumnOrName` but the type for each parameter should be ``` `~pyspark.sql.Column` or str ``` here.
Also, please place the description of each parameter in the next line with indentation like
```
year : class:`~pyspark.sql.Column` or str
    The year to build the date.
```